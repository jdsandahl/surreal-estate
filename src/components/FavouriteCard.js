import React from "react";
import "../styles/FavouriteCard.css";

const FavouriteCard = ({ propertyListing: { title }, userId }) => {
  return (
    <div className="favourite-card">
      <div className="favourite-card__detail">{title}</div>
      <button className="favourite-card__btn --email">Email</button>
      <button className="favourite-card__btn --remove">Remove</button>
    </div>
  );
};

export default FavouriteCard;
