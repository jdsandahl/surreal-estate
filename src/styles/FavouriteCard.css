.favourite-card {
  display: flex;
  background-color: #ffffff;
  margin: 10px;
  padding: 10px;
}

.favourite-card__detail {
  flex: 1 1 auto;
}

.favourite-card__mailto {
  text-decoration: none;
  color: green;
}

.favourite-card__btn {
  flex: 0 1 100px;
  justify-self: flex-end;
  margin: 5px;
}

.favourite-card__btn--email {
  border: 2px solid green;
  background-color: #d1f2de;
}

.favourite-card__btn--remove {
  border: 2px solid red;
  color: red;
  background-color: #fcd3c9;
}
