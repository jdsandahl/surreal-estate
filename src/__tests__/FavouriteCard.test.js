import React from "react";
import { render, fireEvent } from "@testing-library/react";
import FavouriteCard from "../components/FavouriteCard";

describe("FavouriteCard", () => {
  let mockKey;
  let mockFavourite;
  let mockOnRemoveFavourite;

  beforeEach(() => {
    mockKey = "11a2s3d4f5g";
    mockFavourite = {
      title: "test property",
      email: "test@test.com",
    };
    mockOnRemoveFavourite = jest.fn();
  });

  test("it renders correctly against snapshot", () => {
    const { asFragment } = render(
      <FavouriteCard
        key={mockKey}
        propertyListing={mockFavourite}
        onRemoveFavourite={mockOnRemoveFavourite}
      />
    );
    const component = asFragment();

    expect(component).toMatchSnapshot();
  });

  test("it calls the onRemoveFavourites method when Remove is clicked", () => {
    const { getByRole } = render(
      <FavouriteCard
        key={mockKey}
        propertyListing={mockFavourite}
        onRemoveFavourite={mockOnRemoveFavourite}
      />
    );

    const removeFavourite = getByRole("button", { name: /remove/i });
    fireEvent.click(removeFavourite);
    expect(mockOnRemoveFavourite).toHaveBeenCalledTimes(1);
  });
});
