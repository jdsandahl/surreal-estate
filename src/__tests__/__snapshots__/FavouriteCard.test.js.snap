// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouriteCard it renders correctly against snapshot 1`] = `
<DocumentFragment>
  <div
    class="favourite-card"
  >
    <div
      class="favourite-card__detail"
    >
      test property
    </div>
    <button
      class="favourite-card__btn favourite-card__btn--email"
    >
      <a
        class="favourite-card__mailto"
        href="mailto:test@test.com"
      >
        Email
      </a>
    </button>
    <button
      class="favourite-card__btn favourite-card__btn--remove"
      href="#"
    >
      Remove
    </button>
  </div>
</DocumentFragment>
`;
